waku/waku_peer_exchange/protocol.nim

```diff
@@ -169,25 +218,33 @@ proc updatePxEnrCache(wpx: WakuPeerExchange) {.async.} =
 proc initProtocolHandler(wpx: WakuPeerExchange) =
   proc handler(conn: Connection, proto: string) {.async, gcsafe, closure.} =
     var buffer: seq[byte]
-    try:
-      buffer = await conn.readLp(DefaultMaxRpcSize.int)
-    except CatchableError as exc:
-      waku_px_errors.inc(labelValues = [exc.msg])
-      return
-
-    let decBuf = PeerExchangeRpc.decode(buffer)
-    if decBuf.isErr():
-      waku_px_errors.inc(labelValues = [decodeRpcFailure])
-      return
-
-    let rpc = decBuf.get()
-    trace "peer exchange request received"
-    let enrs = wpx.getEnrsFromCache(rpc.request.numPeers)
-    let res = await wpx.respond(enrs, conn)
-    if res.isErr:
-      waku_px_errors.inc(labelValues = [res.error])
-    else:
-      waku_px_peers_sent.inc(enrs.len().int64())
+    wpx.requestRateLimiter.checkUsageLimit(WakuPeerExchangeCodec, conn):
+      try:
+        buffer = await conn.readLp(DefaultMaxRpcSize.int)
+      except CatchableError as exc:
+        waku_px_errors.inc(labelValues = [exc.msg])
+        discard await wpx.respondError(
```

Comment:
Is it possible to better handle these `discard`? This question applies elsewhere were we are discarding things